haskellPackages: update hackage and stackage

sternenseemann opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

More information about Haskell packages in nixpkgs can be found [in the nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#haskell).

---

This is the follow-up to #339272. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

rvl committed:
haskellPackages.dear-imgui: Fix build

rvl committed:
haskellPackages.reflex-sdl2: not broken

rvl committed:
haskellPackages.vivid: Fix build with a small patch

Some library dependency must have stopped re-exporting 'void', so now
it needs an extra import line.

This package doesn't have any VCS repo.

rvl committed:
haskellPackages.gi-gtk-declarative: Fix build

rvl committed:
haskellPackages.reflex-gloss: jailbreak to fix the build

rvl committed:
haskellPackages.reflex-gi-gtk: jailbreak to fix the build

rvl committed:
haskellPackages.pipes-extras: jailbreak to fix the build

I'm not sure why this needs a jailbreak. There is already a hackage
revision to fix it:
https://hackage.haskell.org/package/pipes-extras-1.0.15/revisions/

rvl committed:
haskellPackages.pipes-http: Drop obsolete jailbreak

rvl committed:
haskell.packages.{ghc98,ghc910}.broadcast-chan: Jailbreak

sternenseemann committed:
haskellPackages: stackage LTS 22.33 -> LTS 22.36

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

sternenseemann committed:
all-cabal-hashes: 2024-09-03T10:29:19Z -> 2024-10-05T14:46:54Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

sternenseemann committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
haskellPackages.cloudy: add myself as maintainer

cdepillabout committed:
haskellPackages.cloudy: generate optparse-applicative completions

fgaz committed:
haskellPackages.{gemini-server,language-gemini}: Unmark broken

Fixed in revisions 3 and 2

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

vcunat commented:
I dropped darwin builds from the jobset.  I believe we can put it back when things settle down a bit.

sternenseemann commented:
Feel free to disable evals altogether if necessary; I think no one is available to work on anything at least until the weekend.

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
Partially revert "treewide: replace webkitgtk to webkitgtk_4_0"

This commit partially reverts ccec93c1c748f8e1e879edc42e3562999a3f20d1.

This will break a lot of packages in haskellPackages (due to a missing
webkitgtk), but at least hackage-packages.nix is reproducible with the
generation shell script again.

Unfortunately, it seems the generated nature of hackage-packages.nix was
ignored when preparing the webkitgtk change:
https://github.com/NixOS/nixpkgs/pull/345611#issuecomment-2394993096

sternenseemann committed:
haskellPackages.Cabal_3_12_1_0: keep for cabal-install and friends

sternenseemann commented:
The fallout from #345611 is going to be really annoying to deal with.

sternenseemann committed:
haskell.compiler.ghcHEAD: 9.11.20240410 -> 9.11.20240423

Unfortunately, this is the newest GHC revision we can update to. After
https://gitlab.haskell.org/ghc/ghc/-/commit/145a6477854d4003a07573d5e7ffa0c9a64ae29c,
it becomes necessary to bootstrap with an in tree Cabal version.
Supporting this in nixpkgs is a waste of time since it wouldn't be
necessary for any release version.

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

yuanw committed:
haskellPackages: Enable darwin builds for some packages depending on mesa

update package.yaml

sternenseemann committed:
haskellPackages.cabal2nix-unstable: 2024-05-20 -> 2024-10-17

sternenseemann committed:
haskellPackages.gi-gdk_4: 4.0.8 -> 4.0.9

sternenseemann committed:
haskell.packages.ghc910.extensions: 0.1.0.2 -> 0.1.0.3

sternenseemann committed:
haskell.packages.ghc{8107,902}.Cabal*_3_10_3_0: use process 1.6.24.0

sternenseemann committed:
haskellPackages.Cabal_3_6_3_0: remove at 3.6.3.0

sternenseemann committed:
haskellPackages.directory_1_3_7_1: remove at 1.3.7.1

sternenseemann committed:
haskellPackages.Diff_0_3_4: remove at 0.3.4

The upstream issue for liquidhaskell is resolved. liquidhaskell is
marked broken for an unrelated issue in one of its dependencies.

sternenseemann committed:
haskellPackages.base16-bytestring: remove at 0.1.1.7

sternenseemann committed:
haskellPackages.brittany_0_13_1_2: remove at 0.13.1.2

sternenseemann committed:
taskell: remove at 1.11.4

haskellPackages.brick_0_70_1: remove at 0.70.1

taskell's upstream repository has been archived after three years of
inactivity. There also doesn't seem to be an obvious fork that provides
a suitable, maintained replacement.

No changelog entry since taskell has already been marked as broken in
24.05 from the beginning.

sternenseemann committed:
haskellPackages.basement_0_0_14: remove at 0.0.14

sternenseemann committed:
haskellPackages.dependent-map_0_2_4_0: remove at 0.2.4.0

sternenseemann committed:
haskellPackages.dependent-sum_0_4: remove at 0.4

sternenseemann committed:
haskellPackages.doctest_0_18_2: remove at 0.18.2

sternenseemann committed:
haskellPackages.foundation_0_0_28: remove at 0.0.28

sternenseemann committed:
haskellPackages.ghc-api-compat_8_6: remove at 8.6

sternenseemann committed:
ihaskell: use stackage version of ghc-syntax-highlighter

Seems like Stackage recently has reintroduced ghc-syntax-highlighter to
its set at version 0.0.10.0 which is just what we need.

sternenseemann committed:
haskellPackages.gi-soup_2_4_28: remove at 2.4.28

sternenseemann committed:
haskellPackages.haddock{,-api}_2_23_1: remove at 2.23.1

sternenseemann committed:
haskellPackages.hinotify_0_3_9: remove at 0.3.9

sternenseemann committed:
haskellPackages.hlint_3_2_8: remove at 3.2.8

sternenseemann committed:
haskellPackages.mmorph_1_1_3: remove at 1.1.3

sternenseemann committed:
haskellPackages.hpack_0_36_0: remove at 0.36.0

sternenseemann committed:
haskellPackages.hspec-golden_0_1_0_3: remove at 0.1.0.3

sternenseemann committed:
haskellPackages.network_2_6_3_1: remove at 2.6.3.1

sternenseemann committed:
haskellPackages.lens-aeson_1_1_3: remove at 1.1.3

sternenseemann committed:
haskellPackages.path_0_9_0: remove at 0.9.0

sternenseemann committed:
haskellPackages.tasty-hspec_1_1_6: remove at 1.1.6

sternenseemann committed:
haskellPackages.th-abstraction: drop redundant th-abstraction constraint

Stackage already specifies == 0.5.*

sternenseemann committed:
glirc, haskellPackages.irc-core: unbreak after latest release(s)

sternenseemann committed:
haskellPackages.{sbv_7_13,crackNum_2_4}: remove

These were used by the now removed pkgs.petrinizer.

sternenseemann committed:
haskellPackages.haddock-library_1_7_0: remove at 1.7.0

stylish-cabal does no longer build for GHC 8.6.5 (binary) for unrelated
reasons. Doesn't really make sense to invest further into it.

sternenseemann committed:
haskellPackages: clean up no longer needed `extra-packages` (#349279)

maralorn committed:
git-annex: Update hash

github-actions[bot] committed:
Merge master into haskell-updates

rvl committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

sternenseemann committed:
haskell.compiler.ghc8107: use build libffi during bootstrapping

Backport a patch (which probably addresses a completely unrelated issue
with hadrian and system libffi for GHC >= 9.2.3) that stops Cabal from
handling libffi includes in libraries/ghci. This allows for cc-wrapper
to freely pick the correct libffi which is important when cross
compiling GHC as we need to correctly pick between build and host/target
libffi each time.

This fixes the build of pkgsCross.riscv64.haskell.compiler.ghc8107.

Tested

- haskell.packages.ghc8107.cabal2nix (x86_64-linux, aarch64-linux)
- pkgsCross.riscv64.haskell.compiler.ghc8107 (x86_64-linux)

Could not test darwin because rcodesign is broken at the moment (?).

sternenseemann commented:
9.10 HLS either needs a downgraded extensions or to be built with Cabal >= 3.14.

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
haskellPackages: Assorted build fixes (#342755)

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
Revert "haskell.packages.ghc910.extensions: 0.1.0.2 -> 0.1.0.3"

This reverts commit c3322f4af0e5e6a37acd8a0c70d18c7f5db6164a.

Reason for revert: extensions 0.1.0.3 doesn't support Cabal == 3.12.*.
Dependency conflicts due to extensions seem annoying to deal with.

sternenseemann committed:
haskell.packages.ghc9101: reflect hackage updates

sternenseemann committed:
haskell.packages.ghc9101.{retrie,apply-refact}: rm broken overrides

Neither package support GHC 9.10 currently. Their corresponding plugins
are thus disabled in HLS.

sternenseemann committed:
Merge branch master into haskell-updates

sternenseemann committed:
haskell.compiler.ghc98: 9.8.2 -> 9.8.3

https://www.haskell.org/ghc/download_ghc_9_8_3.html

github-actions[bot] committed:
Merge master into haskell-updates

Aleksanaa committed:
haskellPackages: Enable darwin builds for some packages depending on mesa (#349186)

sternenseemann committed:
haskell.compiler.ghc98: 9.8.3 -> 9.8.2

HLS can't be built yet with 9.8.3, so it doesn't make sense to make it a
default yet. Unfortunately we are still missing a ghc-lib-parser release
that works with ghc 9.8.3.

sternenseemann committed:
haskellPackages.github-app-token: disable network dependent tests

sternenseemann committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1809515](https://hydra.nixos.org/eval/1809515) of nixpkgs commit [e7735af](https://github.com/NixOS/nixpkgs/commits/e7735afbdce5f32dabb78e9393792625ff6d5eaa) as of 2024-10-21 19:40 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

sternenseemann commented:
We've merged the current iteration into `staging-next`, so it will only propagate to `master` after #348827 is merged. This has been done in an effort to reduce the impact on `master` since we haven't been able to build/test any packages on Darwin (due to a lack of darwin build capacity). Merging to master would have meant introducing ~16k darwin rebuilds without any  binary cache behind them.
